### PR TITLE
Cleanup the unit tests and add some unit tests for controllers

### PR DESCRIPTION
**Pull request by LukasReschke**

### CODE REVIEW

**Comment by LukasReschke** [comment]
The failing unit test is unrelated and also exists on master :see_no_evil:

**Comment by DeepDiver1975** [comment]
oh - didn't see this one .... working on the same area https://github.com/owncloud/contacts/pull/891

**Comment by DeepDiver1975** [comment]
let's rebase this one once #891 is merged

**Comment by jancborchardt** [comment]
#891 is merged. @LukasReschke rebase please.

And then review: @babelouest @DeepDiver1975 @tomneedham thx!

**Comment by DeepDiver1975** [comment]
@LukasReschke can I ask you to rebase this please? THX

**Comment by LukasReschke** [comment]
_huge_ fun :speak_no_evil:

**Comment by LukasReschke** [comment]
Strange. Now all tests pass except the ones that @DeepDiver1975 fixed in https://github.com/owncloud/contacts/commit/95528debca0ec7596f9233af92c4502ca69ffc1e.

@DeepDiver1975 Any idea what I fucked up here? :speak_no_evil:

File: .travis.yml [hunk 1]
```diff
@@ -3,28 +3,32 @@ php:
...
   - wget https://raw.githubusercontent.com/owncloud/administration/master/travis-ci/before_install.sh
-  - bash ./before_install.sh contacts $CORE_BRANCH $DB
-  - cd ../core
-  - php occ app:enable contacts
+  - bash ./before_install.sh $APP_NAME $CORE_BRANCH $DB
 
 script:
-  - cd apps/contacts
-  - find . -name \*.php -not -path './l10n/*' -exec php -l "{}" \;
-  - phpunit --configuration tests/phpunit.xml --testsuite unit-tests --coverage-clover clover.xml
```
Comment (DeepDiver1975) [inline .travis.yml:27]: @LukasReschke afaik I did put this in on purpose

--testsuite unit-tests

File: .travis.yml [hunk 2]
```diff
@@ -32,11 +36,12 @@ script:
...
-    env: DB=mysql
-#  - php: 5.4
-#    env: DB=pgsql
-
+    - php: 5.4
+      env: DB=mysql
+    - php: 5.4
+      env: DB=pgsql
+    - php: 5.4
+      env: DB=oracle
```
Comment (DeepDiver1975) [inline .travis.yml]: let's not test oracle here - objections?

File: .travis.yml [hunk 3]
```diff
@@ -3,28 +3,32 @@ php:
   - 5.4
   - 5.5
   - 5.6
-
-branches:
-  only:
-    - master
-    - /^stable\d+(\.\d+)?$/
+  - hhvm
```
Comment (DeepDiver1975) [inline .travis.yml:7]: php7 ???

File: tests/bootstrap.php [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
 
 if(!class_exists('PHPUnit_Framework_TestCase')) {
 	require_once('PHPUnit/Autoload.php');
 }
 
-include_once('lib/testcase.php');
-
 OC_Hook::clear();
-OC_Log::$enabled = true;
-\OCP\Util::connectHook('OCA\Contacts', 'pre_deleteContact', '\OCA\Contacts\Hooks', 'contactDeletion');
```
Comment (DeepDiver1975) [inline tests/bootstrap.php:16]: @LukasReschke please bring this back - this is the reason why travis is failing - THX

File: tests/phpunit.xml [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
-		<testsuite name="unit-tests">
-			<directory suffix="_test.php">lib</directory>
-		</testsuite>
-		<testsuite name="app-tests">
-			<directory suffix="_app.php">lib</directory>
-		</testsuite>
-	</testsuites>
+<?xml version="1.0" encoding="utf-8" ?>
+<phpunit bootstrap="bootstrap.php"
+		 strict="true"
```
Comment (DeepDiver1975) [inline tests/phpunit.xml]: strict was removed on purpose as well :wink:

**Comment by LukasReschke** [comment]
@DeepDiver1975 Done

**Comment by LukasReschke** [comment]
Tests also passing now. Can I haz :cookie: and this merged?

**Comment by DeepDiver1975** [comment]
THX @LukasReschke - let me have a second look at this - THX

**Comment by DeepDiver1975** [comment]
@LukasReschke ooops - rebase :question: :see_no_evil: THX

**Comment by LukasReschke** [comment]
You folks are killing me.

:gun:

**Comment by LukasReschke** [comment]
No idea if everything still works now… :speak_no_evil:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**